src/include/mutex.i

```diff
@@ -251,3 +251,54 @@ __wt_spin_unlock(WT_SESSION_IMPL *session, WT_SPINLOCK *t)
 #error Unknown spinlock type
 
 #endif
+
+/*
+ * __wt_fair_lock --
+ *	Get a lock.
+ */
+static inline int
+__wt_fair_lock(WT_SESSION_IMPL *session, WT_FAIR_LOCK *lock)
+{
+	uint16_t ticket;
+	int pause_cnt;
+
+        WT_UNUSED(session);
+
+	/*
+	 * Possibly wrap: if we have more than 64K lockers waiting, the ticket
+	 * value will wrap and two lockers will simultaneously be granted the
+	 * lock.
+	 */
+	ticket = __wt_atomic_fetch_add16(&lock->waiter, 1);
+	for (pause_cnt = 0; ticket != lock->owner;) {
```

Comment:
@agorrod, what we're relying on in this loop is that the `WT_PAUSE` and/or `__wt_sleep` inside the loop prevent the compiler from caching `lock->owner` in a register.  We could further call that out with `volatile`, but I don't think that is strictly necessary.